src/cogent3/core/new_sequence.py

```diff
@@ -2597,7 +2597,7 @@ class SeqView(SeqViewABC):
     def __init__(
         self,
         *,
-        parent: typing.Union[str, "SeqsDataABC"],
+        parent: StrORBytesORArray,
```

Comment:
note that `SeqDataView` inherits from `SeqView`, and parent is `SeqsDataABC` in that case.

Reply:
I don't think a base class should declare as valid the types for all potential subclasses. Better to redefine `__init__` in the subclass to make it explicit.